dlproject/data/renoir_dataset.py

```diff
@@ -0,0 +1,56 @@
+import math
+import os
+import glob
+import cv2
+import numpy as np
+import tensorflow as tf
+
+
+class RENOIRDatasetSequence(tf.keras.utils.Sequence):
+    def __init__(self, dataset_folder, dataset_type, batch_size, target_size=(5328, 3000)):
+        if dataset_type not in ['Mi3_Aligned', 'T3i_Aligned']:
+            raise ValueError("Invalid dataset type. Choose 'Mi3_Aligned' or 'T3i_Aligned'.")
+
+        self.dataset_folder = dataset_folder
+        self.dataset_type = dataset_type
+        self.batch_size = batch_size
+        self.target_size = target_size
+        self.test_data_paths = self.__get_test_data_paths()
+
+    def __len__(self):
+        return math.ceil(len(self.test_data_paths) / self.batch_size)
+
+    def __getitem__(self, index):
+        low = index * self.batch_size
+        high = min(low + self.batch_size, len(self.test_data_paths))
```

Comment:
Nel caso in cui l'ultimo batch non abbia dimensione `batch_size`. Non necessariamente il numero di immagini è divisibile per `batch_size`.